openmls/src/extensions/mod.rs

```diff
@@ -517,6 +623,87 @@ impl Extension {
+    /// Get a reference to this extension as [`RequiredCapabilitiesExtension`].
+    /// Returns an [`ExtensionError::InvalidExtensionType`] error if called on
+    /// an [`Extension`] that's not a [`RequiredCapabilitiesExtension`].
+    pub fn as_required_capabilities_extension(
+        &self,
+    ) -> Result<&RequiredCapabilitiesExtension, ExtensionError> {
+        match self {
+            Self::RequiredCapabilities(e) => Ok(e),
+            _ => Err(ExtensionError::InvalidExtensionType(
+                "This is not a RequiredCapabilitiesExtension".into(),
+            )),
+        }
+    }
+
+    /// Get a reference to this extension as [`ExternalSendersExtension`].
+    /// Returns an [`ExtensionError::InvalidExtensionType`] error if called on
+    /// an [`Extension`] that's not a [`ExternalSendersExtension`].
+    pub fn as_external_senders_extension(
+        &self,
+    ) -> Result<&ExternalSendersExtension, ExtensionError> {
+        match self {
+            Self::ExternalSenders(e) => Ok(e),
+            _ => Err(ExtensionError::InvalidExtensionType(
+                "This is not an ExternalSendersExtension".into(),
+            )),
+        }
+    }
+
+    /// Returns the [`ExtensionType`]
+    #[inline]
+    pub const fn extension_type(&self) -> ExtensionType {
+        match self {
+            GroupContextExtension::RequiredCapabilities(_) => ExtensionType::RequiredCapabilities,
+            GroupContextExtension::ExternalSenders(_) => ExtensionType::ExternalSenders,
+            GroupContextExtension::Unknown(kind, _) => ExtensionType::Unknown(*kind),
+        }
+    }
+}
+
+impl TryFrom<Extension> for GroupContextExtension {
+    type Error = InvalidExtensionError;
+
+    fn try_from(value: Extension) -> Result<Self, Self::Error> {
+        match value {
+            Extension::RequiredCapabilities(e) => {
+                Ok(GroupContextExtension::RequiredCapabilities(e))
+            }
+            Extension::ExternalSenders(e) => Ok(GroupContextExtension::ExternalSenders(e)),
+            Extension::Unknown(k, e) => Ok(GroupContextExtension::Unknown(k, e)),
+            _ => Err(InvalidExtensionError::IllegalInGroupContext),
+        }
+    }
+}
+
+impl From<GroupContextExtension> for Extension {
+    fn from(value: GroupContextExtension) -> Self {
+        match value {
+            GroupContextExtension::RequiredCapabilities(e) => Extension::RequiredCapabilities(e),
+            GroupContextExtension::ExternalSenders(e) => Extension::ExternalSenders(e),
+            GroupContextExtension::Unknown(t, e) => Extension::Unknown(t, e),
+        }
+    }
+}
+
+impl TryFrom<Extensions<Extension>> for Extensions<GroupContextExtension> {
+    type Error = InvalidExtensionError;
+
+    fn try_from(value: Extensions<Extension>) -> Result<Self, Self::Error> {
+        let group_extensions: Result<Vec<GroupContextExtension>, Self::Error> =
+            value.iter().map(|e| e.clone().try_into()).collect();
```

Comment:
No need to clone here.
```suggestion
            value.unique.into_iter().map(|e| e.try_into()).collect();
```